core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp

```diff
@@ -0,0 +1,570 @@
+    return *this;
+  }
+
+  /** \brief set per thread and per team scratch size for a specific level of
+   * the scratch hierarchy */
+  TeamPolicyInternal& set_scratch_size(int level, PerTeamValue const& per_team,
+                                       PerThreadValue const& per_thread) {
+    m_team_scratch_size[level]   = per_team.value;
+    m_thread_scratch_size[level] = per_thread.value;
+    return *this;
+  }
+
+  using member_type = Kokkos::Impl::SYCLTeamMember;
+
+ protected:
+  template <class ClosureType, class FunctorType, class BlockSizeCallable>
+  int internal_team_size_common(
+      const FunctorType& /*f*/,
+      BlockSizeCallable&& /*block_size_callable*/) const {
+    // FIXME_SYCL
+    Kokkos::abort("Not implemented!");
+    return 0;
+  }
+
+  template <class ClosureType, class FunctorType>
+  int internal_team_size_max(const FunctorType& /*f*/) const {
+    return m_space.impl_internal_space_instance()->m_maxThreadsPerSM;
+  }
+
+  template <class ClosureType, class FunctorType>
+  int internal_team_size_recommended(const FunctorType& /*f*/) const {
+    // FIXME_SYCL improve
+    return m_space.impl_internal_space_instance()->m_maxThreadsPerSM;
+  }
+};
+
+template <typename FunctorType, typename... Properties>
+class ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties...>,
+                  Kokkos::Experimental::SYCL> {
+ public:
+  using Policy = TeamPolicyInternal<Kokkos::Experimental::SYCL, Properties...>;
+  using functor_type = FunctorType;
+  using size_type    = ::Kokkos::Experimental::SYCL::size_type;
+
+ private:
+  using member_type   = typename Policy::member_type;
+  using work_tag      = typename Policy::work_tag;
+  using launch_bounds = typename Policy::launch_bounds;
+
+  FunctorType const m_functor;
+  Policy const m_policy;
+  size_type const m_league_size;
+  int m_team_size;
+  size_type const m_vector_size;
+  int m_shmem_begin;
+  int m_shmem_size;
+  void* m_scratch_ptr[2];
+  int m_scratch_size[2];
+
+  template <typename Functor>
+  void sycl_direct_launch(const Policy& policy, const Functor& functor) const {
+    // Convenience references
+    const Kokkos::Experimental::SYCL& space = policy.space();
+    Kokkos::Experimental::Impl::SYCLInternal& instance =
+        *space.impl_internal_space_instance();
+    sycl::queue& q = *instance.m_queue;
+
+    q.submit([&](sycl::handler& cgh) {
+      const auto shmem_begin  = m_shmem_begin;
+      const auto shmem_size   = m_shmem_size;
+      const auto scratch_size = m_scratch_size[1];
+      auto team_lambda        = [=](sycl::group<1> g) {
+        // FIXME_SYCL Add scratch memory
+        const member_type team_member(nullptr, shmem_begin, shmem_size, nullptr,
+                                      scratch_size, g);
+
+        if constexpr (std::is_same<work_tag, void>::value)
+          functor(team_member);
+        else
+          functor(work_tag(), team_member);
+      };
+      if (m_team_size > 0)
```

Comment:
that doesn't make any sense. Do you mean team size > 1? If you do team_size==0 the functor should never be called.

Reply:
No, `m_team_size==0` means `Kokkos::AUTO` and the team size is left to `SYCL`.